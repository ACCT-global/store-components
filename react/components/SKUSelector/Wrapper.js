import * as React from 'react'
import { ProductContext } from 'vtex.product-context'
import { path, isEmpty } from 'ramda'

import SKUSelector from './index';

export default (props) => {
  const valuesFromContext = React.useContext(ProductContext)

  const skuSelectorProps = () => {
    if (!valuesFromContext || isEmpty(valuesFromContext)) return props

    const { product, selectedItem } = valuesFromContext

    return {
      skuItems: path(['items'], product),
      skuSelected: selectedItem,
      productSlug: path(['linkText'], product),
    }
  }

  return (
    <SKUSelector { ...skuSelectorProps()} />
  )
}