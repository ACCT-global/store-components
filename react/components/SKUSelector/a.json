{
  "itemId": "12",
  "name": "Rosa - G",
  "nameComplete": "Suéter de Tricô Malharia Zanatta Polo Zíper - Rosa - G",
  "complementName": "",
  "ean": "sueter-rosa-g",
  "referenceId": [],
  "measurementUnit": "un",
  "unitMultiplier": 1,
  "modalType": null,
  "isKit": false,
  "images": [],
  "sellers": [],
  "variations": [ // Array of strings insteadof objects
    "Cor",
    "Tamanho"
  ],
  "Cor": "Rosa",
  "Tamanho": "G" // Simple string insteadof an array
}