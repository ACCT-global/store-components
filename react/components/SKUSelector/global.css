.vtex-sku-selector__item {
  font-size: 90%;
  margin-bottom: 5px;
  margin-right: 10px;
  height: 36px;
}

.vtex-sku-selector__item-image {
  padding: 0;
  height: 40px;
  width: 40px;
}

.vtex-sku-selector__item p {
  margin: 0;
  padding-top: 10px;
}

.vtex-sku-selector__bagde {
  font-size: .71429rem;
  background-color: #ff4136;
  border-radius: 50%;
  color: #fff;
  height: 18px;
  line-height: 18px;
  right: -5px;
  text-align: center;
  top: -6px;
  width: 18px;
  position: absolute;
  padding: 3px;
}

.diagonal-cross {
  background-image: linear-gradient(
      to top right,
      transparent 44%,
      silver 48%,
      currentColor 48%,
      currentColor 52%,
      transparent 52%
  );
}

.frame-around {
  bottom: -0.25rem;
  top: -0.25rem;
  left: -0.25rem;
  right: -0.25rem;
}
