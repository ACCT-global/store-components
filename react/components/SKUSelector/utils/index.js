import {
  curry,
  clone,
  values,
  compose,
  map,
  groupBy,
  gt,
  path,
  prop,
  find,
} from 'ramda'

/**
 * Return the maximum sku price
 * @param {array of sku's} items
 */
export const getMaxSkuPrice = items => {
  if (!items) return 0

  return items.reduce((max, sku) => {
    const [
      {
        commertialOffer: { Price },
      },
    ] = sku.sellers

    return Math.max(max, Price)
  })
}

/**
 * Remove the 'https' from the given url
 * @param {string} url
 */
export const stripUrl = url => url.replace(/^https?:/, '')

/**
 * Parse the variations field in the sku object
 * @param {sku} sku
 */
export const parseSku = sku => {
  const result = clone(sku)

  const variations = sku.variations.map(variation => {
    result[variation.name] = variation.values[0]
    return variation.name
  })

  result.variations = variations

  return result
}

const getQuantity = path([
  'sellers',
  '0',
  'commertialOffer',
  'AvailableQuantity',
])

const findItem = a => {
  return find(item => gt(getQuantity(item), 0), a) || a[0]
}

/**
 * Group the sku items by the variation name specified
 */
export const groupItemsByVariation = (name, items) =>
  compose(
    map(findItem),
    values,
    groupBy(prop(name))
  )(items)

/**
 * Verifies if the variation is color
 * @param {string} variation
 */
export const isColor = variation => {
  if (!variation) return false

  return (
    variation.toLowerCase() === 'cor' || variation.toLowerCase() === 'color'
  )
}

/**
 * Choose wich variation will be the main one.
 * @param {Array[string]} variations
 */
export const getMainVariationName = variations => {
  for (let i = 0; i < variations.length; i++) {
    if (isColor(variations[i])) return variations[i]
  }

  return variations[0]
}
