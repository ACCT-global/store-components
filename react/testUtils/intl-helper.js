import React from 'react'
import * as reactTestingLibrary from 'react-testing-library'
import { IntlProvider } from 'react-intl'
import defaultStrings from '../../messages/en-US.json'
const Context = React.createContext({})

export const render = (node, options) => {
  const rendered = reactTestingLibrary.render(
    <IntlProvider messages={defaultStrings} locale="en-US">
      {node}
    </IntlProvider>,
    options
  )

  return {
    ...rendered,
    rerender: newUi =>
      render(newUi, {
        container: rendered.container,
        baseElement: rendered.baseElement,
      }),
  }
}

// re-export everything
export default reactTestingLibrary
