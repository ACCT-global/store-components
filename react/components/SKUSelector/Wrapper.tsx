import React, { useMemo, useEffect } from 'react'
import useProduct from 'vtex.product-context/useProduct'
import { pick } from 'ramda'
import { useProductDispatch } from 'vtex.product-context/ProductDispatchContext'
import {
  MaybeResponsiveInput,
  ResponsiveInput,
  useResponsiveValues,
} from 'vtex.responsive-values'

import SKUSelector from './index'
import { Variations, InitialSelectionType, DisplayMode } from './types'
import { ShowValueForVariation } from './components/SKUSelector'

const useVariations = (skuSpecifications: SkuSpecification[]) => {
  const result = useMemo(() => {
    const variations: Variations = {}
    for (const specification of skuSpecifications) {
      variations[specification.field.name] = specification.values.map(
        value => value.name
      )
    }
    return variations
  }, [skuSpecifications])
  return result
}

interface Props {
  skuItems: ProductItem[]
  skuSelected: ProductItem
  onSKUSelected?: (skuId: string) => void
  maxItems?: number
  seeMoreLabel: string
  hideImpossibleCombinations?: boolean
  showValueNameForImageVariation?: boolean
  showValueForVariation?: ShowValueForVariation
  imageHeight?: MaybeResponsiveInput<number>
  imageWidth?: MaybeResponsiveInput<number>
  thumbnailImage?: string
  visibleVariations?: string[]
  showVariationsLabels?: boolean
  variationsSpacing?: number
  showVariationsErrorMessage?: boolean
  initialSelection?: InitialSelectionType
  displayMode?: MaybeResponsiveInput<DisplayMode>
  sliderDisplayThreshold?: number
  sliderArrowSize?: number
  sliderItemsPerPage?: ResponsiveInput<number>
}

const SKUSelectorWrapper: StorefrontFC<Props> = props => {
  const valuesFromContext = useProduct()
  const dispatch = useProductDispatch()
  const { imageHeight, imageWidth } = useResponsiveValues(
    pick(['imageHeight', 'imageWidth'], props)
  )
  const skuItems =
    props.skuItems != null
      ? props.skuItems
      : valuesFromContext?.product?.items ?? []

  const skuSelected =
    props.skuSelected != null
      ? props.skuSelected
      : valuesFromContext.selectedItem

  const shouldNotShow =
    skuItems.length === 0 ||
    !skuSelected?.variations ||
    skuSelected.variations.length === 0

  const skuSpecifications = valuesFromContext?.product?.skuSpecifications ?? []
  const variations = useVariations(skuSpecifications)

  useEffect(() => {
    if (dispatch) {
      dispatch({
        type: 'SKU_SELECTOR_SET_IS_VISIBLE',
        args: { isVisible: !shouldNotShow },
      })
    }
  }, [shouldNotShow, dispatch])

  if (shouldNotShow || !skuSelected) {
    return null
  }

  let showValueForVariation = ShowValueForVariation.none
  if (props.showValueForVariation) {
    showValueForVariation = props.showValueForVariation
  } else if (props.showValueNameForImageVariation) {
    showValueForVariation = ShowValueForVariation.image
  }

  return (
    <SKUSelector
      skuItems={skuItems}
      variations={variations}
      imageWidth={imageWidth}
      skuSelected={skuSelected}
      maxItems={props.maxItems}
      imageHeight={imageHeight}
      displayMode={props.displayMode}
      seeMoreLabel={props.seeMoreLabel}
      onSKUSelected={props.onSKUSelected}
      thumbnailImage={props.thumbnailImage}
      initialSelection={props.initialSelection}
      variationsSpacing={props.variationsSpacing}
      showValueForVariation={showValueForVariation}
      showVariationsLabels={props.showVariationsLabels}
      hideImpossibleCombinations={props.hideImpossibleCombinations}
      showVariationsErrorMessage={props.showVariationsErrorMessage}
      sliderItemsPerPage={props.sliderItemsPerPage}
      sliderArrowSize={props.sliderArrowSize}
      sliderDisplayThreshold={props.sliderDisplayThreshold}
    />
  )
}

SKUSelectorWrapper.schema = {
  title: 'admin/editor.skuSelector.title',
  description: 'admin/editor.skuSelector.description',
}

export default SKUSelectorWrapper
