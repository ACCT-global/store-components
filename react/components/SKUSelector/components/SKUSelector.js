import PropTypes from 'prop-types'
import React, { useCallback, memo } from 'react'

import Variation from './Variation'
import { variationShape } from '../utils/proptypes'

import styles from '../styles.css'

/** Renders the main and the secondary variation, if it exists. */
const SKUSelector = ({
  onSelectSKU,
  mainVariation,
  secondaryVariation,
  maxSkuPrice,
  alwaysShowSecondary,
  seeMoreLabel,
  maxItems,
}) => {
  if (!mainVariation) return null

  const shouldShowSecondary =
    (alwaysShowSecondary || mainVariation.value) && secondaryVariation.name

  return (
    <div className={styles.skuSelectorContainer}>
      <Variation
        variation={mainVariation}
        onSelectItem={useCallback(skuId => onSelectSKU(true, skuId), [])}
        isSelected={useCallback(sku => sku[mainVariation.name] === mainVariation.value, [mainVariation])}
        maxSkuPrice={maxSkuPrice}
        seeMoreLabel={seeMoreLabel}
        maxItems={maxItems}
      />
      {shouldShowSecondary && (
        <Variation
          variation={secondaryVariation}
          onSelectItem={useCallback(skuId => onSelectSKU(false, skuId), [])}
          isSelected={useCallback(sku => sku.itemId === secondaryVariation.value, [secondaryVariation])}
          maxSkuPrice={maxSkuPrice}
          seeMoreLabel={seeMoreLabel}
          maxItems={maxItems}
        />
      )}
    </div>
  )
}

SKUSelector.propTypes = {
  /** Function to go to the product page of a given sku */
  onSelectSKU: PropTypes.func.isRequired,
  /** Name and list of options of the main variation */
  mainVariation: variationShape,
  /** Name and list of options of the secondary variation */
  secondaryVariation: variationShape,
  /** Max price find on the sku list */
  maxSkuPrice: PropTypes.number.isRequired,
  /** If true, show secondary options (if present), even when main variation is not picked yet */
  alwaysShowSecondary: PropTypes.bool,
  seeMoreLabel: PropTypes.string,
  maxItems: PropTypes.number,
}

export default memo(SKUSelector)
