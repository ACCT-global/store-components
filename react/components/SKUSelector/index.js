import PropTypes from 'prop-types'
import React, { useState, useEffect, useMemo, memo, useCallback } from 'react'
import { useRuntime } from 'vtex.render-runtime'
import { filter, head, isEmpty, compose, keys, length } from 'ramda'
import useProduct from 'vtex.product-context/useProduct'

import SKUSelector from './components/SKUSelector'
import { skuShape } from './utils/proptypes'
import {
  parseSku,
  isColor,
  uniqueOptionToSelect,
  findItemWithSelectedVariations,
} from './utils'

const keyCount = compose(
  length,
  keys
)
const filterSelected = filter(Boolean)

const buildEmptySelectedVariation = variations => {
  const variationNames = Object.keys(variations)
  const result = {}
  for (const variationName of variationNames) {
    result[variationName] = null
  }
  return result
}

/** receives an item and the variations object, returns the selected variations for that item */
const selectedVariationFromItem = (item, variations) => {
  const variationNames = Object.keys(variations)
  const result = {}
  for (const variationName of variationNames) {
    result[variationName] = item[variationName]
  }
  return result
}

const useImagesMap = (items, variations) => {
  return useMemo(() => {
    const variationNames = Object.keys(variations)
    const result = {}
    for (const variationName of variationNames) {
      // Today, only "Color" variation should show image, need to find a more resilient way to tell this, waiting for backend
      if (!isColor(variationName)) {
        continue
      }
      const imageMap = {}
      const variationValues = variations[variationName]
      for (const variationValue of variationValues) {
        const item = items.find(sku => sku[variationName] === variationValue)
        imageMap[variationValue] = item && head(item.images)
      }
      result[variationName] = imageMap
    }
    return result
  }, [items, variations])
}

const useAllSelectedEvent = (selectedVariations, variationsCount) => {
  const { dispatch } = useProduct()
  useEffect(() => {
    if (dispatch && selectedVariations) {
      const selectedNotNull = filterSelected(selectedVariations)
      const selectedCount = keyCount(selectedNotNull)
      const allSelected = selectedCount === variationsCount
      dispatch({
        type: 'SKU_SELECTOR_SET_VARIATIONS_SELECTED',
        allSelected,
      })
    }
  }, [dispatch, selectedVariations, variationsCount])
}

/**
 * Display a list of SKU items of a product and its specifications.
 */
const SKUSelectorContainer = ({
  skuItems = [],
  onSKUSelected,
  seeMoreLabel,
  maxItems,
  variations,
  skuSelected,
  hideImpossibleCombinations,
}) => {
  const variationsCount = keyCount(variations)
  const [selectedSkuId, setSelectedSkuId] = useState(null)
  const [selectedVariations, setSelectedVariations] = useState(null)

  useAllSelectedEvent(selectedVariations, variationsCount)

  const parsedItems = useMemo(() => skuItems.map(parseSku), [skuItems])

  const { setQuery } = useRuntime()
  const redirectToSku = skuId => {
    setQuery(
      { skuId },
      {
        replace: true,
      }
    )
  }

  useEffect(() => {
    const initalVariations = skuSelected
      ? selectedVariationFromItem(parseSku(skuSelected), variations)
      : buildEmptySelectedVariation(variations)
    skuSelected && setSelectedSkuId(skuSelected.itemId)
    setSelectedVariations(initalVariations)
  }, [])

  const imagesMap = useImagesMap(parsedItems, variations)

  const onSelectItem = useCallback(
    ({
      name: variationName,
      value: variationValue,
      skuId,
      isMainAndImpossible,
      possibleItems,
    }) => {
      const isRemoving = selectedVariations[variationName] === variationValue
      const newSelectedVariation = !isMainAndImpossible
        ? {
            ...selectedVariations,
            [variationName]: isRemoving ? null : variationValue,
          }
        : {
            ...buildEmptySelectedVariation(variations),
            [variationName]: variationValue,
          }
      // Set here for a better response to user
      setSelectedVariations(newSelectedVariation)
      const uniqueOptions = isRemoving
        ? {}
        : uniqueOptionToSelect(
            possibleItems,
            newSelectedVariation,
            isMainAndImpossible
          )
      const finalSelected = {
        ...newSelectedVariation,
        ...uniqueOptions,
      }
      if (!isEmpty(uniqueOptions)) {
        setSelectedVariations(finalSelected)
      }

      const selectedNotNull = filterSelected(finalSelected)
      const selectedCount = keyCount(selectedNotNull)
      const allSelected = selectedCount === variationsCount
      let skuIdToRedirect = skuId
      if (!skuIdToRedirect || !isEmpty(uniqueOptions)) {
        const newItem = findItemWithSelectedVariations(
          possibleItems,
          finalSelected
        )
        skuIdToRedirect = newItem.itemId
      }

      if (selectedSkuId === skuIdToRedirect || isRemoving) {
        // If it is still the same SKU or we are removing a selection, no need to do anything.
        return
      }

      if (onSKUSelected) {
        setSelectedSkuId(skuIdToRedirect)
        onSKUSelected(skuIdToRedirect)
      } else {
        if (allSelected || isColor(variationName)) {
          setSelectedSkuId(skuIdToRedirect)
          // Use setTimeout so the UI responsiveness on mobile is better, redirect causes heavy processing.
          setTimeout(() => redirectToSku(skuIdToRedirect), 0)
        }
      }
    },
    [selectedVariations, variations, onSKUSelected]
  )

  if (!selectedVariations) {
    return null
  }

  return (
    <SKUSelector
      variations={variations}
      seeMoreLabel={seeMoreLabel}
      maxItems={maxItems}
      skuItems={parsedItems}
      selectedVariations={selectedVariations}
      imagesMap={imagesMap}
      onSelectItem={onSelectItem}
      hideImpossibleCombinations={hideImpossibleCombinations}
    />
  )
}

SKUSelectorContainer.propTypes = {
  /** SKU selected */
  skuSelected: skuShape,
  /** List of SKU Items */
  skuItems: PropTypes.arrayOf(skuShape).isRequired,
  /** Callback that is called when an SKU is selected */
  onSKUSelected: PropTypes.func,
  seeMoreLabel: PropTypes.string,
  maxItems: PropTypes.number,

  /** Object with dynamic keys, with keys being the name of variations and its values being an array of possible values.
   * Example: { "size": ["small", "medium", "large"], "color": ["blue", "yellow"] }
   */
  variations: PropTypes.object,

  hideImpossibleCombinations: PropTypes.bool,
}

SKUSelectorContainer.defaultProps = {
  maxItems: 10,
  hideImpossibleCombinations: true,
}

export default memo(SKUSelectorContainer)
