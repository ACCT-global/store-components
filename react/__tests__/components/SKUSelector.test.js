import React from 'react'
import { renderWithIntl } from 'intl-helper'

import SKUSelector from './../../SKUSelector'
import SKU from 'sku-helper'

describe('<SKUSelector />', () => {
  const renderComponent = customProps => {
    const props = {
      skuSelected: SKU(),
      skuItems: [SKU('Black'), SKU('Blue'), SKU('Yellow')],
      ...customProps,
    }
    return renderWithIntl(<SKUSelector {...props} />)
  }

  it('should be mount', () => {
    expect(renderComponent()).toBeDefined()
  })

  it('should match the snapshot', () => {
    expect(renderComponent()).toMatchSnapshot()
  })
})
