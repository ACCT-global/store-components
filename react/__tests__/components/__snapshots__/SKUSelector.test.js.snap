// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SKUSelector /> should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="skuSelectorContainer"
  >
    <div
      class="skuSelectorSubcontainer flex flex-column mb7"
    >
      <div
        class="skuSelectorNameContainer ma1"
      >
        <span
          class="skuSelectorName c-muted-2 db t-small overflow-hidden mb3"
        >
          Color
        </span>
        <div
          class="inline-flex flex-wrap ml2"
        >
          <div
            class="skuSelectorItem relative di pointer flex items-center skuSelectorItemImage"
            role="button"
            tabindex="0"
          >
            <div
              class="absolute frameAround b--action-primary br3 bw1"
            />
            <div
              class="w-100 h-100 ba br2 b b--muted-4 z-1 c-muted-5 flex items-center overflow-hidden hover-b--muted-2"
            >
              <div
                class="absolute absolute--fill"
              />
              <div
                class=""
              >
                <img
                  alt=""
                  src="sku-image-Black-url"
                />
              </div>
            </div>
          </div>
          <div
            class="skuSelectorItem relative di pointer flex items-center skuSelectorItemImage"
            role="button"
            tabindex="0"
          >
            <div
              class="absolute frameAround b--action-primary br3 bw1"
            />
            <div
              class="w-100 h-100 ba br2 b b--muted-4 z-1 c-muted-5 flex items-center overflow-hidden hover-b--muted-2"
            >
              <div
                class="absolute absolute--fill"
              />
              <div
                class=""
              >
                <img
                  alt=""
                  src="sku-image-Blue-url"
                />
              </div>
            </div>
          </div>
          <div
            class="skuSelectorItem relative di pointer flex items-center skuSelectorItemImage"
            role="button"
            tabindex="0"
          >
            <div
              class="absolute frameAround b--action-primary br3 bw1"
            />
            <div
              class="w-100 h-100 ba br2 b b--muted-4 z-1 c-muted-5 flex items-center overflow-hidden hover-b--muted-2"
            >
              <div
                class="absolute absolute--fill"
              />
              <div
                class=""
              >
                <img
                  alt=""
                  src="sku-image-Yellow-url"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<SKUSelector /> should render only three main variations 1`] = `
<DocumentFragment>
  <div
    class="skuSelectorContainer"
  >
    <div
      class="skuSelectorSubcontainer flex flex-column mb7"
    >
      <div
        class="skuSelectorNameContainer ma1"
      >
        <span
          class="skuSelectorName c-muted-2 db t-small overflow-hidden mb3"
        >
          Color
        </span>
        <div
          class="inline-flex flex-wrap ml2"
        >
          <div
            class="skuSelectorItem relative di pointer flex items-center skuSelectorItemImage"
            role="button"
            tabindex="0"
          >
            <div
              class="absolute frameAround b--action-primary br3 bw1 ba"
            />
            <div
              class="w-100 h-100 ba br2 b b--muted-4 z-1 c-muted-5 flex items-center overflow-hidden"
            >
              <div
                class="absolute absolute--fill diagonalCross"
              />
              <div
                class=""
              >
                <span
                  class="c-on-base t-body"
                >
                  Gray
                </span>
              </div>
            </div>
          </div>
          <div
            class="skuSelectorItem relative di pointer flex items-center skuSelectorItemImage"
            role="button"
            tabindex="0"
          >
            <div
              class="absolute frameAround b--action-primary br3 bw1"
            />
            <div
              class="w-100 h-100 ba br2 b b--muted-4 z-1 c-muted-5 flex items-center overflow-hidden hover-b--muted-2"
            >
              <div
                class="absolute absolute--fill diagonalCross"
              />
              <div
                class=""
              >
                <span
                  class="c-on-base t-body"
                >
                  Black
                </span>
              </div>
            </div>
          </div>
          <div
            class="skuSelectorItem relative di pointer flex items-center skuSelectorItemImage"
            role="button"
            tabindex="0"
          >
            <div
              class="absolute frameAround b--action-primary br3 bw1"
            />
            <div
              class="w-100 h-100 ba br2 b b--muted-4 z-1 c-muted-5 flex items-center overflow-hidden hover-b--muted-2"
            >
              <div
                class="absolute absolute--fill diagonalCross"
              />
              <div
                class=""
              >
                <span
                  class="c-on-base t-body"
                >
                  Blue
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="skuSelectorSubcontainer flex flex-column mb7"
    >
      <div
        class="skuSelectorNameContainer ma1"
      >
        <span
          class="skuSelectorName c-muted-2 db t-small overflow-hidden mb3"
        >
          Size
        </span>
        <div
          class="inline-flex flex-wrap ml2"
        >
          <div
            class="skuSelectorItem relative di pointer flex items-center"
            role="button"
            tabindex="0"
          >
            <div
              class="absolute frameAround b--action-primary br3 bw1 ba"
            />
            <div
              class="w-100 h-100 ba br2 b b--muted-4 z-1 c-muted-5 flex items-center overflow-hidden"
            >
              <div
                class="absolute absolute--fill diagonalCross"
              />
              <div
                class="c-on-base center pl5 pr5 z-1"
              >
                <span
                  class="c-on-base t-body"
                >
                  1
                </span>
              </div>
            </div>
          </div>
          <div
            class="skuSelectorItem relative di pointer flex items-center"
            role="button"
            tabindex="0"
          >
            <div
              class="absolute frameAround b--action-primary br3 bw1"
            />
            <div
              class="w-100 h-100 ba br2 b b--muted-4 z-1 c-muted-5 flex items-center overflow-hidden hover-b--muted-2"
            >
              <div
                class="absolute absolute--fill diagonalCross"
              />
              <div
                class="c-on-base center pl5 pr5 z-1"
              >
                <span
                  class="c-on-base t-body"
                >
                  2
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
